/*
  CarStandFCOOL API

  An API for managing a cloud-based car dealership service

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package openapitools.services.meetingservice;

import "google/protobuf/empty.proto";
import public "models/meeting.proto";

service MeetingService {
  rpc MeetingsCreate (MeetingsCreateRequest) returns (Meeting);

  rpc MeetingsDelete (MeetingsDeleteRequest) returns (google.protobuf.Empty);

  rpc MeetingsReadAll (google.protobuf.Empty) returns (MeetingsReadAllResponse);

  rpc MeetingsReadOne (MeetingsReadOneRequest) returns (Meeting);

  rpc MeetingsUpdate (MeetingsUpdateRequest) returns (Meeting);

}

message MeetingsCreateRequest {
  // Meeting to create
  Meeting meeting = 1;

}

message MeetingsDeleteRequest {
  // ID of the meeting.
  int32 meetingId = 1;

}

message MeetingsReadAllResponse {
  repeated Meeting data = 1;
}

message MeetingsReadOneRequest {
  // ID of the meeting.
  int32 meetingId = 1;

}

message MeetingsUpdateRequest {
  // ID of the meeting.
  int32 meetingId = 1;
  Meeting meeting = 2;

}

